src/typechecker.rs

```diff
@@ -2446,8 +2451,16 @@ pub fn typecheck_statement(
                     .current_function_index
                     .map(|i| project.functions[i].return_type_id),
             );
+            error = error.or(err);
+
+            if project.inside_defer {
+                error = error.or(Some(JaktError::TypecheckError(
+                    "`return` is not allowed inside `defer`".to_string(),
```

Comment:
```suggestion
                    "'return' is not allowed inside 'defer'".to_string(),
```

Reply:
(I don't think having markdown in compiler error messages is a good choice, people will inevitably copy and paste that into github and accidentally alter the string, possibly breaking search and all that :))

Reply:
Yeah, I copied what GCC does and used the nifty open/close single quote characters.